frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRendererComponent.java

```diff
@@ -103,6 +106,23 @@ public BranchOrCommitCellRendererComponent(
     int textPadding = calculateTextPadding(graphTable, maxGraphNodePositionInRow);
     appendTextPadding(textPadding);
 
+    if (graphItem.isBranchItem() && graphItem.asBranchItem().isCurrentBranch()) {
+      var gitMacheteRepositorySnapshot = graphTable.getGitMacheteRepositorySnapshot();
+      if (gitMacheteRepositorySnapshot != null) {
+        var ongoingRepositoryOperation = gitMacheteRepositorySnapshot.getOngoingRepositoryOperation();
+        if (ongoingRepositoryOperation != OngoingRepositoryOperation.NO_OPERATION) {
+          var ongoingOperationName = Match(ongoingRepositoryOperation).of(
+              Case($(OngoingRepositoryOperation.CHERRY_PICK), "CHERRY-PICK"),
+              Case($(OngoingRepositoryOperation.MERGING), "MERGING"),
+              Case($(OngoingRepositoryOperation.REBASING), "REBASING"),
+              Case($(OngoingRepositoryOperation.REVERTING), "REVERTING"),
+              Case($(), ""));
+          SimpleTextAttributes attributes = SimpleTextAttributes.ERROR_ATTRIBUTES;
+          append(ongoingOperationName + CELL_TEXT_FRAGMENTS_SPACING, attributes);
```

Comment:
Hmm do we want to append the spacing even if `ongoingOperationName` is empty? :thinking:

Reply:
It's likely to be not a case - this `Match` construction cover all possible cases, so `Case($(), ""))` is only for safety and should never happen.

Reply:
Aaah ok, right